web-share/share-empty.https.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share no known fields</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="resources/manual-helper.js"></script>
+  </head>
+  <body>
+    <script>
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share());
+        }, 'share with no arguments (same as empty dictionary)');
+
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share({}));
```

Comment:
@domenic I don't understand why there is friction in using a dictionary for a set of optional arguments with a "must provide at least one" condition. You say there are no better tools available... what would hypothetically such tools look like? It seems perfectly fine to use a dictionary for this.

If you were designing an API in any other programming language, you'd make a struct or class, make each field optional, and add a check that at least one is supplied. There is no need to invent a new programming language construct to express this concept. (The only friction is the "optional" part but I'll continue that discussion on heycam/webidl#130.)

The real question I have here (and I think Marcos has too) is with regards to extensibility... is it OK to plan to add a new field to the dictionary later, which in the future will be valid to use on its own (e.g., `{image: x}`) but against older versions of the spec use of that on its own would result in a `TypeError`? The idea being that you never do that unless you first do feature detection to ensure it is supported.